python/ray/includes/gcs_client.pxi

```diff
@@ -606,6 +606,68 @@ cdef class InnerGcsClient:
 
         return (is_accepted, rejection_reason_message.decode())
 
+    #############################################################
+    # Publisher methods
+    #############################################################
+
+    def publish_error(self, key_id: bytes, error_type: str, message: str,
+                      job_id: Optional[JobID] = None, timeout = None):
+        cdef:
+            CErrorTableData error_info
+            c_string c_key_id = key_id
+            int64_t timeout_ms = round(1000 * timeout) if timeout else -1
+
+        if job_id is None:
+            job_id = ray.JobID.nil()
+        error_info.set_job_id(job_id.binary())
+        error_info.set_type(error_type)
+        error_info.set_error_message(message)
+        error_info.set_timestamp(time.time())
+
+        with nogil:
+            check_status_timeout_as_rpc_error(
+                self.inner.get().Publisher().PublishError(
+                    move(c_key_id), move(error_info), timeout_ms))
+
+    def publish_logs(self, log_json: dict, timeout = None):
```

Comment:
yikes.. do you know if there are existing tests for this logic somewhere?

Reply:
There is `test_gcs_pubsub.py`.